aiohomekit/controller/ble/pairing.py

```diff
@@ -102,17 +107,54 @@
 }
 BLE_AID = 1  # The aid for BLE devices is always 1
 
+ENABLE_BROADCAST_PAYLOAD = TLV.encode_list(
+    [
+        (
+            HAP_BLE_CHARACTERISTIC_CONFIGURATION_REQUEST_TLV.kTLVHAPParamCharacteristicConfigurationProperties,
+            int.to_bytes(1, 2, "little"),
+        ),
+        (
+            HAP_BLE_CHARACTERISTIC_CONFIGURATION_REQUEST_TLV.kTLVHAPParamCharacteristicConfigurationBroadcastInterval,
+            bytes([0x01]),
+        ),
+    ]
+)
+
+GENERATE_BROADCAST_KEY_PAYLOAD = (
+    bytes([HAP_BLE_PROTOCOL_CONFIGURATION_REQUEST_TLV.GenerateBroadcastEncryptionKey])
+    + b"\x00"
+)
+
 WrapFuncType = TypeVar("WrapFuncType", bound=Callable[..., Any])
 
 
 def operation_lock(func: WrapFuncType) -> WrapFuncType:
     """Define a wrapper to only allow a single operation at a time."""
 
-    async def _async_wrap(self: BlePairing, *args: Any, **kwargs: Any) -> None:
+    async def _async_operation_lock_wrap(
+        self: BlePairing, *args: Any, **kwargs: Any
+    ) -> None:
         async with self._operation_lock:
             return await func(self, *args, **kwargs)
 
-    return cast(WrapFuncType, _async_wrap)
+    return cast(WrapFuncType, _async_operation_lock_wrap)
+
+
+def restore_connection_and_resume(func: WrapFuncType) -> WrapFuncType:
+    """Define a wrapper restore connection, populate data, and then resume when the operation completes."""
+
+    async def _async_restore_and_resume(
+        self: BlePairing, *args: Any, **kwargs: Any
+    ) -> None:
+        """Restore connection, populate data, and then resume when the operation completes."""
+        await self._populate_accessories_and_characteristics()
+        try:
+            return await func(self, *args, **kwargs)
```

Comment:
Previously we did this on a call later, but sometimes the operation would take longer than the timer so we now restore everything after operation is done so if they ask to turn on a light and set a color it doesn't turn on, restore subscriptions, and then set the color because the timer fires in the middle